### DISCUSSION TITLE
Show sub-properties as columns?

### DISCUSSION THREAD

**Discussion by MarredCheese** [datalust/seq-tickets#1859] [General]
I love how I can click on a property and add it as a column.  But I don't see any way to do it for sub-properties, and because of that, I find myself on the verge of dirtying up my code by flattening out the logged properties, as shown below.  But that can't be right, because "structured logging" is one of the main selling points of Serilog, right?  I know I can get at sub-properties via queries, but I'd much rather stick to the log view and be able to see the new events stream in as they happen and be able to click on events to expand them and get more detail.

For example, is there any way to add `@Exception.Message` as a column?  Then I could keep doing this:

```csharp
logger.Error(ex, "Reporting production failed");
```

Instead of having to make it redundant like this.

```csharp
logger
    .ForContext("ExceptionMessage", ex.Message)
    .Error(ex, "Reporting production failed");

```
The situation gets much worse if I have my own object full of properties that I might want to show as a column.  Before:

```csharp
using (LogContext.PushProperty("ReportProductionParams", param, true))
```

After:

```csharp
using (LogContext.PushProperty("Part", param.Part))
using (LogContext.PushProperty("Lot", param.Lot))
using (LogContext.PushProperty("ProductionOrder", param.ProductionOrder))
using (LogContext.PushProperty("Location", param.Location))
...
```

**Comment by nblumhardt** [comment]
Hi! Thanks for dropping us a line.

Sub-properties can be added as columns in Seq 2023.1 - you may need to update, if you're not already using that version:

![image](https://github.com/datalust/seq-tickets/assets/342712/500cd142-0733-40b3-82a8-6cee5c037e9c)

The `@Exception.Message` case is different, since Seq doesn't treat exceptions as structured objects; in that case, adding an `ExceptionMessage` property is still the way to go.

Let me know if this helps,
Nick

**Comment by MarredCheese** [comment]
Sweet!  I'll upgrade to the latest version.  Thanks!
